runtimes/protocol/identity-management.ts

```diff
@@ -0,0 +1,22 @@
+import { ProtocolRequestType } from './lsp'
+
+export type ListProfilesParams = {}
+export type ListProfilesResult = {
+    profiles?: any[]
+    ssoSessions?: any[]
+}
+export type ListProfilesError = {}
+export const listProfilesRequestType = new ProtocolRequestType<
+    ListProfilesParams,
+    ListProfilesResult,
+    never,
+    ListProfilesError,
+    void
+>('aws/identity/listProfiles')
```

Comment:
If we include the profile-related calls (ListProfiles and UpdateProfile, maybe more in the future) in the identity runtime feature rather than a separate profile feature, it will need to throw a not implemented error in browser runtimes.  

While we could theoretically introduce the concept of profiles in browsers and store in a browser-based storage rather than shared config, that is an entirely separate discussion out of scope for this project.

Reply:
> While we could theoretically introduce the concept of profiles in browsers and store in a browser-based storage rather than shared config, that is an entirely separate discussion out of scope for this project.

that's the idea here: in the runtimes package we define the protocol and server interface, and it's up to the server implementation whether it uses `WebStorage` or `FileSystem` or anything else. With that being said, ideally, server implementations should be runtime-agnostic with actual runtime-specific features injected into them via common interface.

Reply:
I did a little exploration in the space of having the runtime package implement the runtime-specific file access regarding shared config.  There were some challenges such as the paths are specific (~/.aws/config & credentials) and having to access environment variables which the browser cannot do either.  Given that was just requirements for shared config and I'm sure other file/host access requirements will differ, I gave up on that approach and went back to implementing file access in the server for now.  The implementation does model native storage vs web storage so they can be swapped out within the server at least.